app/src/androidTest/java/com/nervousfish/nervousfish/test/ContactActivitySteps.java

```diff
@@ -18,24 +21,39 @@
 import com.nervousfish.nervousfish.data_objects.IKey;
 import com.nervousfish.nervousfish.data_objects.SimpleKey;
 import com.nervousfish.nervousfish.service_locator.EntryActivity;
+import com.nervousfish.nervousfish.service_locator.IServiceLocator;
 
 import org.hamcrest.Description;
 import org.hamcrest.Matcher;
 import org.junit.internal.matchers.TypeSafeMatcher;
 
+import java.io.IOException;
+import java.util.List;
+
 import cucumber.api.CucumberOptions;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.Then;
 import cucumber.api.java.en.When;
 
 import static android.support.test.espresso.Espresso.onView;
 import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.action.ViewActions.scrollTo;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isClickable;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.core.AllOf.allOf;
 
 @CucumberOptions(features = "features")
 public class ContactActivitySteps extends ActivityInstrumentationTestCase2<EntryActivity> {
 
-    private Activity parent = null;
+    private Contact contact = null;
+    private IServiceLocator serviceLocator = null;
+    private final static String TEST_NAME = "TestPerson";
+    private final static IKey TEST_KEY = new SimpleKey("my key", "key");
+    private final static String DIFFERENT_NAME = "DifferentName";
```

Comment:
Variables like this one should be set via the `.feature` file, that way you can also easily add test for different values